### PR TITLE
Deny designs save with empty service name from Meshery Extensions

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: server/models/pattern/core/pattern.go [hunk 1]
```diff
@@ -317,6 +317,9 @@ func processCytoElementsWithPattern(eles []cytoscapejs.Element, pf *Pattern, cal
 		if err := json.Unmarshal(svcByt, &svc); err != nil {
 			return fmt.Errorf("failed to create service from the metadata in the scratch")
 		}
+		if svc.Name == "" {
+			return fmt.Errorf("cannot save service with empty name")
```
Comment (leecalcote) [inline server/models/pattern/core/pattern.go:321]: What Meshery error code belongs to this error? Suggested remediation?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: server/models/pattern/core/pattern.go (hunk 1 above)
Comment (Revolyssup) [reply server/models/pattern/core/pattern.go:321]: This internal error is returned and wrapped into a meshklt error where the function is called. `ErrSavePattern`

Comment (leecalcote) [reply server/models/pattern/core/pattern.go:321]: Good deal. Although - https://docs.meshery.io/search/?q=ErrSavePattern. ☹️

Comment (leecalcote) [reply server/models/pattern/core/pattern.go:321]: I see "ErrSavePatternCode", but its description is not helpful in this situation. https://docs.meshery.io/reference/error-codes

Comment (leecalcote) [reply server/models/pattern/core/pattern.go:321]: @acald-creator We need to continue our audit of error codes.

**Comment by leecalcote** [comment]
Pending creation of a new error code.

**Comment by Revolyssup** [comment]
> Pending creation of a new error code.

Added a new error code for this specific function.

**Comment by leecalcote** [comment]
With an eye toward MeshModel, will this error code fall to the wayside?

**Review by leecalcote — APPROVED** [review]
LGTM. Will you please open an issue or two regarding improving our error codes? One of the issues should be to combine the moniker and the code into the same column on the docs page.

**Pull request merged**